### PR TITLE
Create WorkerDao implementation on jpa

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Adds JPA WorkerDao implementation

### What issues does this PR fix or reference?
#394 

### Tests written?
Yes

### Docs requirements?
None

### CODE REVIEW

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/JpaWorkerDao.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Jpa implementation of {@link WorkerDao},
+ *
+ * @author Max Shaposhnik
+ */
+public class JpaWorkerDao implements WorkerDao {
```
Comment (evoevodin) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/JpaWorkerDao.java]: it should be singleton

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/JpaWorkerDao.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<WorkerImpl> getWorkers(String workspaceId) throws ServerException {
+        requireNonNull(workspaceId, "Workspace identifier required");
+        try {
+            return managerProvider.get()
+                                  .createQuery("SELECT worker " +
```
Comment (evoevodin) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/JpaWorkerDao.java]: What do you think about moving this query to the `@NamedQueries` section on top of `WorkspaceImpl` ? 
The [example](https://github.com/eclipse/che/blob/9d5e335ccc57edf9b04bd26005ee43eaad2e0197/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/WorkspaceImpl.java#L54-L63).

**Comment by skabashnyuk** [comment]
ok

**Comment by mshaposhnik** [comment]
Added listeners for cascading removal. @evoevodin pls review

**Comment by evoevodin** [comment]
Looks good! Go merge

**Pull request merged**